src/pages/index.js

```diff
@@ -0,0 +1,393 @@
+
+// Form elements
+const editModal = document.querySelector('#edit-modal');
+const editFormElement = editModal.querySelector('.modal__form');
+const editModalCloseBtn = editModal.querySelector('.modal__close-btn');
+const editModalNameInput = editModal.querySelector('#profile-name-input');
+const editModalDescriptionInput = editModal.querySelector(
+  '#profile-description-input'
+);
+
+// Card Modal elements
+const cardModal = document.querySelector('#add-card-modal');
+const cardForm = cardModal.querySelector('.modal__form');
+const cardSubmitBtn = cardModal.querySelector('.modal__submit-btn');
+const cardNameInput = cardModal.querySelector('#add-card-name-input');
+const cardLinkInput = cardModal.querySelector('#add-card-link-input');
+const cardModalCloseBtn = cardModal.querySelector('.modal__close-btn');
+
+// Preview Modal elements
+const previewModal = document.querySelector('#preview-modal');
+const previewModalImageEl = previewModal.querySelector('.modal__image');
+const previewModalCaptionEl = previewModal.querySelector('.modal__caption');
+const previewModalCloseBtn = previewModal.querySelector('.modal__close-btn');
+
+//Card related elements
+const cardTemplate = document.querySelector('#card-template');
+const cardsList = document.querySelector('.cards__list');
+
+// select the avatar modal and its elements
+const avatarModal = document.querySelector('#avatar-modal');
+const avatarForm = avatarModal.querySelector('.modal__form');
+const avatarSubmitBtn = avatarModal.querySelector('.modal__submit-btn');
+const avatarModalCloseBtn = avatarModal.querySelector('.modal__close-btn');
+const avatarInput = avatarModal.querySelector('#profile-avatar-input');
+
+// Delete Modal elements
+const deleteModal = document.querySelector('#delete-modal');
+const deleteForm = deleteModal.querySelector('.modal__form');
+const deleteModalCancelBtn = deleteModal.querySelector('#delete-cancel-btn');
+let selectedCard;
+let selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add('modal_opened');
+  document.addEventListener('keydown', handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove('modal_opened');
+  document.removeEventListener('keydown', handleEscapeKey);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  // editModalSubmitBtn.textContent = 'Saving...';
+  // change text content to "Saving..."
+  const editModalSubmitBtn = evt.submitter;
+  editModalSubmitBtn.textContent = 'Saving...';
+
+  setButtonText(editModalSubmitBtn, true, "Save", "Saving...");
+  // disable the button
+  api.editUserInfo({ name: editModalNameInput.value, about: editModalDescriptionInput.value })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+
+      // todo - call setButtonText instead
+      editModalSubmitBtn.textContent = "Save";
```

Comment:
Please, reuse the `setButtonText` function here, to avoid code duplication


```js

    .finally(() => {
      setButtonText(editModalSubmitBtn, false);  // like this - with false
    });
```